zapcore/json_encoder.go

```diff
@@ -193,51 +195,61 @@ func (enc *jsonEncoder) AddUint8(k string, v uint8)         { enc.AddUint64(k, u
 func (enc *jsonEncoder) AddUintptr(k string, v uintptr)     { enc.AddUint64(k, uint64(v)) }
 func (enc *jsonEncoder) AppendByte(v byte)                  { enc.AppendUint8(uint8(v)) }
 func (enc *jsonEncoder) AppendComplex64(v complex64)        { enc.AppendComplex128(complex128(v)) }
+func (enc *jsonEncoder) AppendDuration(val time.Duration)   { enc.DurationFormatter(val, enc) }
 func (enc *jsonEncoder) AppendFloat64(v float64)            { enc.appendFloat(v, 64) }
 func (enc *jsonEncoder) AppendFloat32(v float32)            { enc.appendFloat(float64(v), 32) }
 func (enc *jsonEncoder) AppendInt(v int)                    { enc.AppendInt64(int64(v)) }
 func (enc *jsonEncoder) AppendInt32(v int32)                { enc.AppendInt64(int64(v)) }
 func (enc *jsonEncoder) AppendInt16(v int16)                { enc.AppendInt64(int64(v)) }
 func (enc *jsonEncoder) AppendInt8(v int8)                  { enc.AppendInt64(int64(v)) }
 func (enc *jsonEncoder) AppendRune(v rune)                  { enc.AppendInt32(int32(v)) }
+func (enc *jsonEncoder) AppendTime(val time.Time)           { enc.TimeFormatter(val, enc) }
 func (enc *jsonEncoder) AppendUint(v uint)                  { enc.AppendUint64(uint64(v)) }
 func (enc *jsonEncoder) AppendUint32(v uint32)              { enc.AppendUint64(uint64(v)) }
 func (enc *jsonEncoder) AppendUint16(v uint16)              { enc.AppendUint64(uint64(v)) }
 func (enc *jsonEncoder) AppendUint8(v uint8)                { enc.AppendUint64(uint64(v)) }
 func (enc *jsonEncoder) AppendUintptr(v uintptr)            { enc.AppendUint64(uint64(v)) }
 
 func (enc *jsonEncoder) Clone() Encoder {
-	clone := &jsonEncoder{JSONConfig: enc.JSONConfig}
+	clone := &jsonEncoder{EncoderConfig: enc.EncoderConfig}
 	clone.bytes = append(buffers.Get(), enc.bytes...)
 	return clone
 }
 
 func (enc *jsonEncoder) EncodeEntry(ent Entry, fields []Field) ([]byte, error) {
 	final := &jsonEncoder{
-		JSONConfig: enc.JSONConfig,
-		bytes:      buffers.Get(),
+		EncoderConfig: enc.EncoderConfig,
+		bytes:         buffers.Get(),
 	}
 
 	final.bytes = append(final.bytes, '{')
-	final.LevelFormatter(ent.Level).AddTo(final)
-	final.TimeFormatter(ent.Time).AddTo(final)
-	final.NameFormatter(ent.LoggerName).AddTo(final)
-	if ent.Caller.Defined {
+	if final.LevelKey != "" {
+		final.addKey(final.LevelKey)
+		final.LevelFormatter(ent.Level, final)
```

Comment:
nit: any of `FormatLevel`, `EncodeLevel`, or `AppendLevel` would fit better as the internal field name.